internal/supervisor/supervisor.go

```diff
@@ -0,0 +1,111 @@
+package supervisor
+
+import (
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/mattermost/awat/internal/store"
+	"github.com/mattermost/awat/internal/translator"
+	"github.com/mattermost/awat/model"
+)
+
+// Supervisor is responsible for scheduling and launching Translations
+// in series
+type Supervisor struct {
+	logger  log.FieldLogger
+	store   *store.SQLStore
+	bucket  string
+	workdir string
+}
+
+// NewSupervisor returns a Supervisor prepared with the needed
+// metadata to operate
+func NewSupervisor(store *store.SQLStore, logger log.FieldLogger, bucket, workdir string) *Supervisor {
+	return &Supervisor{
+		store:   store,
+		logger:  logger.WithField("supervisor", model.NewID()),
+		bucket:  bucket,
+		workdir: workdir,
+	}
+}
+
+// Start runs the Supervisor's main routine on a new goroutine both
+// periodically and forever
+func (s *Supervisor) Start() {
+	s.logger.Info("Supervisor started")
+	go func() {
+		for {
+			s.supervise()
+			time.Sleep(60 * time.Second)
+		}
+	}()
+}
+
+// supervise queries the database for available Translations and
+// works through the batch returned serially
+func (s *Supervisor) supervise() {
+	work, err := s.store.GetTranslationsReadyToStart()
+	if err != nil {
+		s.logger.WithError(err).Error("Failed to query database for pending translations")
+		return
+	}
+
+	if len(work) > 0 {
+		s.logger.Debugf("Found %d requests pending to be translated", len(work))
+	}
+
+	for _, translation := range work {
+		s.logger.Debugf("Translating %s for Installation %s...", translation.ID, translation.InstallationID)
+		// TODO XXX expose the Pod name as an env var and use it as the second argument here
+		err = s.store.TryLockTranslation(translation, model.NewID())
+		if err != nil {
+			s.logger.WithError(err).Warnf("failed to lock Translation %s", translation.ID)
+			continue
+		}
+		defer s.store.UnlockTranslation(translation)
```

Comment:
Not sure if it is intended but this will cause translation to be unlocked only after all `work` have been processed (after the loop finishes and function exits) which means this instance may at some point hold locks for all translations.

Reply:
Nope, not intended, good eye. You're absolutely right.

Reply:
Fixing the race resolved below by returning 1 Translation instead of 10 also resolves this